File: leetcode/1281.cpp
// Runtime: 0 ms (Beats 100%)
// Memory: 6 MB (Beats 6.7%)
class Solution {
 public:
  int subtractProductAndSum(int n) {
    int digits_sum = 0, digits_product = 1;
    while (n) {
      digits_sum += n % 10;
      digits_product *= n % 10;
      n /= 10;
    }
    return digits_product - digits_sum;
  }
};

File: leetcode/1281.py
"""
Runtime: 32 ms, faster than 59.70% of Python3 online submissions for Subtract the Product and Sum of Digits of an Integer.
Memory Usage: 14.3 MB, less than 41.30% of Python3 online submissions for Subtract the Product and Sum of Digits of an Integer.
"""
from functools import reduce


class Solution:
    def subtractProductAndSum(self, n: int) -> int:
        return reduce(lambda x, y: x * y, [int(j) for j in str(n)]) - sum(
            [int(j) for j in str(n)]
        )
